ssas/service/public/oktalive_test.go

```diff
@@ -0,0 +1,184 @@
+// +build okta
+
+// To enable this test suite, either:
+
+// - Run from your IDE after setting the env vars below
+//   OR
+// - Run "DATABASE_URL=postgresql://postgres:toor@127.0.0.1:5432/bcda?sslmode=disable go test -tags=okta -v" from the ssas/service/public directory
+
+package public
+
+import (
+	"encoding/hex"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/pborman/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/CMSgov/bcda-app/ssas/okta"
+)
+
+type OktaLiveTestSuite struct {
+	suite.Suite
+	oc *OktaMFAPlugin
+	email string
+	userId string
+	smsFactorId string
+}
+
+func (s *OktaLiveTestSuite) SetupSuite() {
+	s.email = os.Getenv("OKTA_MFA_EMAIL")
+	s.userId = os.Getenv("OKTA_MFA_USER_ID")
+	s.smsFactorId = os.Getenv("OKTA_MFA_SMS_FACTOR_ID")
+
+	if s.email == "" || s.userId == "" || s.smsFactorId == "" {
+		s.FailNow(fmt.Sprintf("Cannot run live Okta tests without env vars set: OKTA_MFA_EMAIL=%s; OKTA_MFA_USER_ID=%s; OKTA_MFA_SMS_FACTOR_ID=%s", s.email, s.userId, s.smsFactorId))
+	}
+}
+
+func (s *OktaLiveTestSuite) SetupTest() {
+	s.oc = NewOktaMFA(okta.Client())
+}
+
+func (s *OktaLiveTestSuite) TestPostFactorChallengeSuccess() {
+	trackingId := uuid.NewRandom().String()
+	factor := Factor{Id: s.smsFactorId, Type: "sms"}
+
+	factorVerification, err := s.oc.postFactorChallenge(s.userId, factor, trackingId)
+	if err != nil || factorVerification == nil {
+		s.FailNow("factor result not parsed: " + err.Error())
```

Comment:
Thank you for putting messages on `FailNow()` calls! It's nice when a test actually reports why it failed. (And I am reminded to do that in tests I write!)